protocols/floodsub/src/protocol.rs

```diff
@@ -56,10 +62,16 @@ where
     type Error = FloodsubDecodeError;
     type Future = Pin<Box<dyn Future<Output = Result<Self::Output, Self::Error>> + Send>>;
 
-    fn upgrade_inbound(self, mut socket: TSocket, _: Self::Info) -> Self::Future {
+    fn upgrade_inbound(self, socket: TSocket, _: Self::Info) -> Self::Future {
         Box::pin(async move {
-            let packet = upgrade::read_length_prefixed(&mut socket, 2048).await?;
-            let rpc = rpc_proto::Rpc::decode(&packet[..]).map_err(DecodeError)?;
+            let mut framed = Framed::<TSocket, prost_codec::Codec<rpc_proto::Rpc>>::new(
+                socket,
+                prost_codec::Codec::new(MAX_MESSAGE_LEN_BYTES),
+            );
+            let rpc = match framed.next().await {
+                Some(v) => v.unwrap_or_default(),
```

Comment:
We can't do unwrap-or-default here, but we should map the error up. Multiple possibilities:

- Use `try_next()?` instead of `next`
- Match on the result: `Some(Ok(v))` and have another match branch `Some(Err(e))`.

`try_next` is probably cleaner.